packages/py-api/py_api/controllers/hackathon/participants_controller.py

```diff
@@ -241,71 +150,151 @@ def add_participant(cls, participant: NewParticipant, jwt_token: str | None = No
 
             # Find the teams which can accept a new participant
             for team in random_teams:
-                if (len(team.team_members) < 6):
-                    return cls.add_participant_to_existing_random_team(team, participant)
+                if len(team.team_members) < 6:
+                    return await cls.add_participant_to_existing_team(team, participant)
 
             # If all the random teams are full, check if there is space for creating a new one
-            if (TeamFunctionality.get_count_of_teams() < 15):
-                return cls.add_participant_to_new_random_team(participant)
+            if TeamFunctionality.get_count_of_teams() < 15:
+                return await cls.add_participant_to_new_team(participant, generate_random_team=True)
 
-            else:
-                return JSONResponse(content={"message": "The maximum number of teams is reached."}, status_code=409)
+            return JSONResponse(content={"message": "The maximum number of teams is reached."}, status_code=409)
 
     @classmethod
-    def add_participant_to_existing_random_team(cls, available_team: HackathonTeam, participant: NewParticipant) -> JSONResponse:
+    async def handle_registration_via_invite_link(cls, jwt_token: str, participant: NewParticipant) -> JSONResponse:
+        result = JWTFunctionality.decode_token(jwt_token, is_invite=True)
+        if isinstance(result, dict):
+            decoded_token = result
+        else:
+            return JSONResponse(content=result[0], status_code=result[1])
+
+        team = TeamFunctionality.fetch_team(
+            team_name=decoded_token.get("team_name"),
+        )
+        if not team:
+            return JSONResponse(content={"message": "Team with this name does not exist"}, status_code=404)
+
+        if len(team.team_members) < 6:
+            return await cls.add_participant_to_existing_team(team, participant, is_invite=True)
+
+        return JSONResponse(content={"message": "The maximum number of teams is reached."}, status_code=409)
+
+    @classmethod
+    async def add_participant_to_existing_team(
+            cls, existing_team: HackathonTeam,
+            participant: NewParticipant, is_invite: bool = False,
+    ) -> JSONResponse:
         try:
             # Creates new participant
-            new_participant = ParticipantsFunctionality.create_participant(
+            new_participant = ParticipantsFunctionality.insert_participant(
                 participant,
             )
+            new_participant_object_id = str(new_participant.inserted_id)
 
-            new_participant_object_id = str(
-                new_participant.inserted_id,
-            )
-            # Adds the participant to the team
```

Comment:
This comment was defining which part of the workflow we're currently in -> grouping the add_participant_to_team_object and update_participant as they have to go hand in hand.